src/Microsoft.ML.AutoML/AutoMLExperiment/AutoMLExperiment.cs

```diff
@@ -0,0 +1,262 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.AutoML
+{
+    internal class AutoMLExperiment
+    {
+        private readonly AutoMLExperimentSettings _settings;
+        private readonly MLContext _context;
+        private double _bestError = double.MaxValue;
+        private TrialResult _bestTrialResult = null;
+        private readonly IServiceCollection _serviceCollection;
+
+        public AutoMLExperiment(MLContext context, AutoMLExperimentSettings settings)
+        {
+            this._context = context;
+            this._settings = settings;
+            this._serviceCollection = new ServiceCollection();
+        }
+
+        private void InitializeServiceCollection()
+        {
+            this._serviceCollection.TryAddSingleton(this._context);
+            this._serviceCollection.TryAddSingleton(this._settings);
+            this._serviceCollection.TryAddSingleton<IMonitor, MLContextMonitor>();
+            this._serviceCollection.TryAddSingleton<ITrialRunnerFactory, TrialRunnerFactory>();
+            this._serviceCollection.TryAddSingleton<ITunerFactory, CfoTunerFactory>();
+            this._serviceCollection.TryAddTransient<BinaryClassificationCVRunner>();
+            this._serviceCollection.TryAddTransient<BinaryClassificationTrainTestRunner>();
+            this._serviceCollection.TryAddTransient<RegressionTrainTestRunner>();
+            this._serviceCollection.TryAddTransient<RegressionCVRunner>();
+            this._serviceCollection.TryAddTransient<MultiClassificationCVRunner>();
+            this._serviceCollection.TryAddTransient<MultiClassificationTrainTestRunner>();
+            this._serviceCollection.TryAddScoped<HyperParameterProposer>();
+            this._serviceCollection.TryAddScoped<PipelineProposer>();
+        }
+
+        public AutoMLExperiment SetTrainingTimeInSeconds(int trainingTimeInSeconds)
+        {
+            this._settings.MaxExperimentTimeInSeconds = (uint)trainingTimeInSeconds;
+            return this;
+        }
```

Comment:
Is there a reason you don't just take a uint as the parameter instead of needing to cast it? If someone passes in a negative value here I think it will throw an error.

Reply:
Oh nope,, I would fix the parameter by using a uint instead, thanks for finding it out